### PR TITLE
Fallback to cmake_modules if Eigen3 not found

### PR DESCRIPTION
**Pull request by jmtatsch**
for the time being as proposed in http://wiki.ros.org/jade/Migration#Eigen_CMake_Module_in_cmake_modules

### CODE REVIEW

File: pcl_ros/CMakeLists.txt [hunk 1]
```diff
@@ -4,7 +4,15 @@ project(pcl_ros)
...
 find_package(Boost REQUIRED COMPONENTS system filesystem thread)
-find_package(Eigen3 REQUIRED)
+find_package(Eigen3)
+if(NOT EIGEN3_FOUND)
+  # Fallback to cmake_modules
+  find_package(cmake_modules REQUIRED)
+  find_package(Eigen REQUIRED)
+  set(EIGEN3_INCLUDE_DIRS ${EIGEN_INCLUDE_DIRS})
+  set(EIGEN3_LIBRARIES ${EIGEN_LIBRARIES})
+  # Possibly map additional variables to the EIGEN3_ prefix.
```
Comment (wjwwood) [inline pcl_ros/CMakeLists.txt]: This comment is probably not need here.

**Comment by wjwwood** [comment]
Lgtm, I'll let @paulbovbel review and merge, etc.

**Comment by jmtatsch** [comment]
Since I installed Eigen 3.2.6 this workaround fails with

```
CMake Error at /opt/ros/jade/share/catkin/cmake/catkin_package.cmake:159 (message):
  catkin_package() DEPENDS on 'Eigen3' which must be find_package()-ed
  before.  If it is a catkin package it can be declared as CATKIN_DEPENDS
  instead without find_package()-ing it.
Call Stack (most recent call first):
  /opt/ros/jade/share/catkin/cmake/catkin_package.cmake:98 (_catkin_package)
  CMakeLists.txt:64 (catkin_package)
```

**Comment by jmtatsch** [comment]
Using pkgconfig as proposed in https://github.com/ros-perception/perception_pcl/issues/106 works fine for me.

**Comment by wjwwood** [comment]
@jmtatsch it fails becuase you have to either set the `Eigen3_FOUND` variable to the value of `Eigen_FOUND` (in the case that you always give `catkin_package()` the `Eigen3` name) or you have to conditionally pass the correct name (either `Eigen` or `Eigen3`) to `catkin_package()`.

**Comment by paulbovbel** [comment]
@wjwwood, given this issue with OSX, would you reconsider deprecating the FindEigen.cmake in cmake_modules?

**Comment by wjwwood** [comment]
@paulbovbel Which issue?

**Comment by paulbovbel** [comment]
Sorry, #106. The issue being that FindEigen3.cmake is not provided by Eigen in OSX, or Arch.

**Comment by wjwwood** [comment]
@paulbovbel I believe OS X has one now: https://github.com/Homebrew/homebrew/issues/35225

As for Arch, I still think it would be best for them to fix their packages, since the upstream Eigen3 release provides the module.

**Comment by paulbovbel** [comment]
My mistake, I thought this was related. @jmtatsch , what's the motivation for this PR?

**Comment by jmtatsch** [comment]
@paulbovbel The motivation for this PR is simply that pcl_ros does (still) not find the current Eigen on OS X El Capitan without it. Sure it would be preferred if `find_package(Eigen3 REQUIRED)` simply worked as expected on all platforms.

**Comment by wjwwood** [comment]
@jmtatsch You should be trying `find_package(Eigen)` first, since that's what is distributed with Eigen. If that doesn't work then it's a bug of the Homebrew formulae and we should file an issue.

**Comment by wjwwood** [comment]
Sorry, that should be `find_package(Eigen3)`. The `find_package(Eigen)` is from our `cmake_modules`.

**Comment by jmtatsch** [comment]
Still an open issue as of now, here is the relevant home-brew issue https://github.com/Homebrew/homebrew/issues/44937

**Comment by paulbovbel** [comment]
Apologies for the delay.

I'm extrapolating from https://github.com/ros-planning/moveit/commit/a6ba84fb98369c3710fb8a4e0adb76e211d42961 that this has been resolved. However, not having any experience with homebrew/OSX, i'm happy to take a patch if necessary.

**Pull request closed without merging**